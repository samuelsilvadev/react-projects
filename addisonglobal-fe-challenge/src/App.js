import React from 'react';
import { Route, Switch } from 'react-router-dom';
import Home from './Home';

import Header from './components/header/Header';

import './App.scss';

const App = () => (
	<>
		<Header />
		<Switch>
			<Route exact path="/" component={Home} />
		</Switch>
	</>
);

export default App;
