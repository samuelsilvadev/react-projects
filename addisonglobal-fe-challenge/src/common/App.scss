* {
	box-sizing: border-box;
	margin: 0;
	padding: 0;
}

body {
	margin: 0;
	padding: 0;
	font-family: sans-serif;
}

.main {
	padding: 0 15px;
}
