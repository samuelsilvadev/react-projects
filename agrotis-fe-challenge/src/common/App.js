import React from 'react';
import styled from 'styled-components';
import { Route, Switch } from 'react-router-dom';

import { Header, HEADER_HEIGHT } from './components';
import Home from './pages/Home';
import { GlobalStyle } from './Global.style';
import { gteSmallMedia } from './media.style';

const Main = styled.main`
	background-color: #ecf0f1;
	min-height: calc(100vh - ${HEADER_HEIGHT});
	padding: 2rem;

	${gteSmallMedia} {
		padding: 2rem 4rem;
	}
`;

const App = () => (
	<>
		<GlobalStyle />
		<Header />
		<Main>
			<Switch>
				<Route exact path="/" component={Home} />
			</Switch>
		</Main>
	</>
);

export default App;
