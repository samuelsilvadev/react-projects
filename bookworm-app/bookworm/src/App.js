import React from 'react';
import { Route } from 'react-router-dom';
import PropTypes from 'prop-types';
import HomePage from './components/pages/HomePage'
import LoginPage from './components/pages/LoginPage';
import DashBoardPage from './components/pages/DashBoardPage';
import UserRoute from './components/routes/UserRoute';
import GuestRoute from './components/routes/GuestRoute';
import SignUp from './components/pages/SignUpPage';

const App = ({ location }) => (
  <div className="ui container">
    <Route location={location} path="/" exact component={HomePage} />
    <GuestRoute location={location} path="/login" component={LoginPage} />
    <GuestRoute location={location} path="/signup" component={SignUp} />
    <UserRoute location={location} path="/dashboard" component={DashBoardPage} />
  </div>
);

App.propTypes = {
  location: PropTypes.shape({
    pathname: PropTypes.string.isRequired,
  }).isRequired
};

export default App;
