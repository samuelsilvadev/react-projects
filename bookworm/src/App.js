import React from 'react';

const App = () => (
  <div>Hi</div>
);

export default App;
