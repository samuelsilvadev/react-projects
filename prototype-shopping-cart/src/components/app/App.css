.shoppingCart {
    max-width: 500px;
    margin: auto;
    width: 100%;
}

.shoppingCart__title {
    padding-bottom: 1rem;
    text-align: center;
}
