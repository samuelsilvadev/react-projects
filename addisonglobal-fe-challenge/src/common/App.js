import React from 'react';
import { Route, Switch } from 'react-router-dom';
import { Home } from './pages';

import Header from './components/header/Header';
import BetSlip from './components/betslip';

import { useAsideContext, withAsideContext } from './contexts/AsideContext';

import './App.scss';

const App = () => {
	const { handleOpen, handleClose, isAsideVisible } = useAsideContext();

	return (
		<>
			<Header onAsideOpen={handleOpen} />
			<main className="main">
				<Switch>
					<Route exact path="/" component={Home} />
				</Switch>
				<BetSlip open={isAsideVisible} onClose={handleClose} />
			</main>
		</>
	);
};

export default withAsideContext(App);
