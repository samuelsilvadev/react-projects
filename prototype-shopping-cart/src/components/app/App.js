import React from 'react';
import CounterContainer from '../counter/CounterContainer';

import './App.css';

const mockCounters = [
	{
		id: 1,
		value: 2,
	},
	{
		id: 2,
	},
	{
		id: 3,
	}
];

class App extends React.Component {

	state = {
		counters: mockCounters,
	}

	render() {
		const { counters } = this.state;
		
		return (
			<section className="shoppingCart">
				<h1 className="shoppingCart__title">Shopping Cart</h1>
				<CounterContainer counters={counters} onDelete={this._removeCounter} />
			</section>
		);
	}

	_removeCounter = (id) => {
		const { counters } = this.state;
		const newCounters = counters.filter(counter => counter.id !== id);

		this.setState({
			counters: newCounters,
		});
	}
}

export default App;
