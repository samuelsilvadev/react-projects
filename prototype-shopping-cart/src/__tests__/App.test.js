import React from 'react';
import { shallow } from 'enzyme';

import App from './../components/app/App';

describe('<App />', () => {
    it('should render correctly and match snapshot', () => {
        const AppComponent = shallow(<App />);

        expect(AppComponent).toMatchSnapshot();
    });
});
