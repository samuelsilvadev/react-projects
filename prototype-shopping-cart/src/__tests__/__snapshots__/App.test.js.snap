// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should render correctly and match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1
          className="shoppingCart__title"
        >
          Shopping Cart
        </h1>,
        <CounterContainer
          counters={
            Array [
              Object {
                "id": 1,
                "value": 2,
              },
              Object {
                "id": 2,
              },
              Object {
                "id": 3,
              },
            ]
          }
          onDelete={[Function]}
        />,
      ],
      "className": "shoppingCart",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Shopping Cart",
          "className": "shoppingCart__title",
        },
        "ref": null,
        "rendered": "Shopping Cart",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "counters": Array [
            Object {
              "id": 1,
              "value": 2,
            },
            Object {
              "id": 2,
            },
            Object {
              "id": 3,
            },
          ],
          "onDelete": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            className="shoppingCart__title"
          >
            Shopping Cart
          </h1>,
          <CounterContainer
            counters={
              Array [
                Object {
                  "id": 1,
                  "value": 2,
                },
                Object {
                  "id": 2,
                },
                Object {
                  "id": 3,
                },
              ]
            }
            onDelete={[Function]}
          />,
        ],
        "className": "shoppingCart",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Shopping Cart",
            "className": "shoppingCart__title",
          },
          "ref": null,
          "rendered": "Shopping Cart",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "counters": Array [
              Object {
                "id": 1,
                "value": 2,
              },
              Object {
                "id": 2,
              },
              Object {
                "id": 3,
              },
            ],
            "onDelete": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
